### PR TITLE
Allow conversions between native u types and U128

### PR DESCRIPTION
**Pull request by SwayStar123**
## Description
Adds impl From<u8,u16,u32,u64> for U128

Partially addresses #5797 

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by SwayStar123** [comment]
Blocked by #5826

File: sway-lib-std/src/u128.sw [hunk 1]
```diff
@@ -19,6 +20,98 @@ pub struct U128 {
...
+    /// Casts a `u8` to a `U128`.
+    ///
+    /// # Returns
+    ///
+    /// * [U128] - The `U128` representation of the `u8` value.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    ///
```
Comment (bitzoic) [inline sway-lib-std/src/u128.sw:33]: The examples are missing imports

File: sway-lib-std/src/u128.sw [hunk 2]
```diff
@@ -19,6 +20,98 @@ pub struct U128 {
     lower: u64,
 }
 
+impl From<u8> for U128 {
+    /// Casts a `u8` to a `U128`.
```
Comment (bitzoic) [inline sway-lib-std/src/u128.sw:24]:
```suggestion
    /// Converts a `u8` to a `U128`.
```

**Review by bitzoic — CHANGES_REQUESTED** [review]

File: sway-lib-std/src/u128.sw [hunk 3]
```diff
@@ -19,6 +20,102 @@ pub struct U128 {
...
+    fn from(val: u16) -> Self {
+        Self {
+            upper: 0,
+            lower: val.into(),
+        }
+    }
+}
+
+impl From<u32> for U128 {
+    /// Casts a `u32` to a `U128`.
```
Comment (K1-R1) [inline sway-lib-std/src/u128.sw:72]:
```suggestion
    /// Converts a `u32` to a `U128`.
```

**Review by K1-R1 — CHANGES_REQUESTED** [review]

**Review by K1-R1 — APPROVED** [review]

**Review by bitzoic — APPROVED** [review]

**Pull request merged**